src/openrct2/core/String.cpp

```diff
@@ -343,66 +324,41 @@ namespace String
         return buffer;
     }
 
-    utf8* Format(const utf8* format, ...)
+    u8string StdFormat(const utf8* format, ...)
     {
         va_list args;
         va_start(args, format);
-        utf8* result = Format_VA(format, args);
+        auto result = Format_VA(format, args);
         va_end(args);
         return result;
     }
 
-    utf8* Format_VA(const utf8* format, va_list args)
+    u8string Format_VA(const utf8* format, va_list args)
     {
-        va_list args1, args2;
-        va_copy(args1, args);
-        va_copy(args2, args);
+        va_list copy;
 
-        // Try to format to a initial buffer, enlarge if not big enough
-        size_t bufferSize = 4096;
-        utf8* buffer = Memory::Allocate<utf8>(bufferSize);
+        va_copy(copy, args);
 
-        // Start with initial buffer
-        int32_t len = vsnprintf(buffer, bufferSize, format, args);
-        if (len < 0)
-        {
-            Memory::Free(buffer);
-            va_end(args1);
-            va_end(args2);
+        // Find the required buffer length
+        const int32_t len = vsnprintf(nullptr, 0, format, copy);
 
-            // An error occurred...
-            return nullptr;
-        }
+        va_end(copy);
 
-        size_t requiredSize = static_cast<size_t>(len) + 1;
-        if (requiredSize > bufferSize)
+        if (len >= 0)
         {
-            // Try again with bigger buffer
-            buffer = Memory::Reallocate<utf8>(buffer, bufferSize);
-            len = vsnprintf(buffer, bufferSize, format, args);
-            if (len < 0)
-            {
-                Memory::Free(buffer);
-                va_end(args1);
-                va_end(args2);
+            // Create a buffer that is of the required length
+            std::string buffer(std::size_t(len) + 1, '\0');
 
-                // An error occurred...
-                return nullptr;
-            }
-        }
-        else
-        {
-            // Reduce buffer size to only what was required
-            bufferSize = requiredSize;
-            buffer = Memory::Reallocate<utf8>(buffer, bufferSize);
-        }
+            vsnprintf(buffer.data(), buffer.size(), format, args);
```

Comment:
`args` is being used here, but it is not yet initialized by `va_start`. The call to `va_copy` before does not initialize args, but initializes the copy.

I'll just work on this myself to get this merged. The constant back and forth with reviews is no way to get things in cleanly.